### ISSUE TITLE
Review and revise the process of saving info when deleting a gloss or media.

### ISSUE THREAD

**Issue by vanlummelhuizen** [Signbank/Global-signbank#659]
The code below may need some revision because of the use of LemmaIDGlossTranslation and AnnotationIDGlossTranslation. 

https://github.com/Signbank/Global-signbank/blob/853bc4f736b2143097109b0414461c1da12e0461/signbank/dictionary/models.py#L1788-L1810

**Comment by susanodd** [comment]
@vanlummelhuizen are you working on this issue? If not, I will work on it.

This one is also related to #440. That has evolved into Other media paths and filenames.

**Comment by vanlummelhuizen** [comment]
@susanodd I am not working on this issue so please do. I think the main thing is that idgloss and annotation_idgloss have a many-to-one relationship with Gloss while the code above only takes a default one.

**Comment by susanodd** [comment]
Hmmm. What is the intention of the DeletedGlossOrMedia history?
It would seem the Dataset would also be useful?
The code that creates objects of this model is:
https://github.com/Signbank/Global-signbank/blob/d218da656e6a7d02dd547a227efc6f4c90c637aa/signbank/dictionary/views.py#L2145
https://github.com/Signbank/Global-signbank/blob/d218da656e6a7d02dd547a227efc6f4c90c637aa/signbank/dictionary/models.py#L1871
https://github.com/Signbank/Global-signbank/blob/d218da656e6a7d02dd547a227efc6f4c90c637aa/signbank/video/views.py#L71

But the actual image or gloss gets deleted. The video gets copied to a .bak file or overwritten.

A deleted gloss cannot be restored from a DeletedGlossOrMedia object. There is no information about phonology, etc.
It would need to be restored using a backup database or an exported csv as well.

**Comment by vanlummelhuizen** [comment]
> do I need to add a migration?

@susanodd Yes, I believe changing the choices argument in a model field entails making a new migration. @Woseseltops and I decide to hold of new migrations until the whole FieldChoice revision is merged with master. So, if possible, do not make any migrations.

> A deleted gloss cannot be restored from a DeletedGlossOrMedia object. There is no information about phonology, etc.
> It would need to be restored using a backup database or an exported csv as well.

I believe that DeletedGlossOrMedia is not meant for undoing deletion, but just for having some info about what was deleted. Perhaps @ocrasborn can say something about this?

**Comment by susanodd** [comment]
I added an admin class DeletedGlossOrMediaAdmin to be able to browse things that have been deleted.
The three options gloss, image, and video are all present.
The annotations seem to be okay.

**Comment by susanodd** [comment]
@vanlummelhuizen there is a class `GlossVideoHistory` that gets data filled into it, but is not used any place else in the code.

**Comment by susanodd** [comment]
Glosses are now archived. The delete and restore are saved in the Gloss Revision History

**Comment by susanodd** [comment]
This might be relevant again. I think we need to remove data from this table, as well as make it accessible in the management pages since it is also collecting data for the API functions.

**Comment by vanlummelhuizen** [comment]
> This might be relevant again. I think we need to remove data from this table, as well as make it accessible in the management pages since it is also collecting data for the API functions.

Is this linked to a feature request of bug report? If so, please add references. If not, it probably has very low priority.

**Comment by susanodd** [comment]
I was busy looking through issues to see what has been completed and can be closed (#1367 )

In the code for videos which we have been developing in the past month, there is code that is saving this information.

We don't actually do anything with this saved information.

Unfortunately it doesn't actually save backups or anything, so I'm wondering if this is needed or needs to be revised to make it useful, or make it accessible?

**Comment by vanlummelhuizen** [comment]
I agree that it is, because it is unused, pretty useless. However, we probably should leave it as is, for now. It is not causing any problems and we probably should spend our time on other things.
